ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+        return OGRERR_FAILURE;
+    }
+
+    OGRFieldDefn* fieldDefn = featureDefn_->GetFieldDefn(field);
+    CPLString clmName = launderColumnNames_
+                            ? LaunderName(newFieldDefn->GetNameRef())
+                            : CPLString(newFieldDefn->GetNameRef());
+
+    try
+    {
+        if ((flagsIn & ALTER_NAME_FLAG)
+            && (strcmp(fieldDefn->GetNameRef(), newFieldDefn->GetNameRef())
+                != 0))
+        {
+            CPLString sql = StringFormat(
+                "RENAME COLUMN %s TO %s",
+                GetFullColumnNameQuoted(
+                    schemaName_, tableName_, fieldDefn->GetNameRef())
+                    .c_str(),
+                QuotedIdentifier(clmName).c_str());
+            dataSource_->ExecuteSQL(sql.c_str());
+        }
+
+        if ((flagsIn & ALTER_TYPE_FLAG)
+            || (flagsIn & ALTER_WIDTH_PRECISION_FLAG)
+            || (flagsIn & ALTER_NULLABLE_FLAG)
+            || (flagsIn & ALTER_DEFAULT_FLAG))
+        {
+            ColumnTypeInfo columnTypeInfo = GetColumnTypeInfo(*newFieldDefn);
+            CPLString fieldTypeDef = GetColumnDefinition(columnTypeInfo);
+            if ((flagsIn & ALTER_NULLABLE_FLAG)
+                && fieldDefn->IsNullable() != newFieldDefn->IsNullable())
+            {
+                if (fieldDefn->IsNullable())
+                    fieldTypeDef += " NULL";
+                else
+                    fieldTypeDef += " NOT NULL";
+            }
+
+            if ((flagsIn & ALTER_DEFAULT_FLAG)
+                && ((fieldDefn->GetDefault() == nullptr
+                     && newFieldDefn->GetDefault() != nullptr)
+                    || (fieldDefn->GetDefault() != nullptr
+                        && newFieldDefn->GetDefault() == nullptr)
+                    || (fieldDefn->GetDefault() != nullptr
+                        && newFieldDefn->GetDefault() != nullptr
+                        && strcmp(
+                               fieldDefn->GetDefault(),
+                               newFieldDefn->GetDefault())
+                               != 0)))
+            {
+                fieldTypeDef +=
+                    " DEFAULT "
+                    + ((fieldDefn->GetType() == OFTString)
+                           ? Literal(newFieldDefn->GetDefault())
+                           : CPLString(newFieldDefn->GetDefault()));
+            }
+
+            CPLString sql = StringFormat(
+                "ALTER TABLE %s ALTER(%s %s)",
+                GetFullTableNameQuoted(schemaName_, tableName_).c_str(),
+                QuotedIdentifier(clmName).c_str(), fieldTypeDef.c_str());
+
+            dataSource_->ExecuteSQL(sql.c_str());
+        }
+    }
+    catch (const odbc::Exception& ex)
+    {
+        CPLError(
+            CE_Failure, CPLE_AppDefined, "Failed to alter field %s: %s",
+            fieldDefn->GetNameRef(), ex.what());
+        return OGRERR_FAILURE;
+    }
+
+    // TODO change an entry in attrColumns_;
```

Comment:
TODO to do

Reply:
The TODO is completed in 02a586c3f62ed6ef53a5d3ec00233d73acf4e3fb.